app/screens/edit_profile/edit_profile.tsx

```diff
@@ -0,0 +1,431 @@
+    const theme = useTheme();
+    const keyboardAwareRef = useRef<KeyboardAwareScrollView>();
+    const firstNameRef = useRef<FloatingTextInputRef>(null);
+    const lastNameRef = useRef<FloatingTextInputRef>(null);
+    const usernameRef = useRef<FloatingTextInputRef>(null);
+    const emailRef = useRef<FloatingTextInputRef>(null);
+    const nicknameRef = useRef<FloatingTextInputRef>(null);
+    const positionRef = useRef<FloatingTextInputRef>(null);
+
+    const [userInfo, setUserInfo] = useState<UserInfo>({
+        email: currentUser.email,
+        firstName: currentUser.firstName,
+        lastName: currentUser.lastName,
+        nickname: currentUser.nickname,
+        position: currentUser.position,
+        username: currentUser.username,
+    });
+    const [canSave, setCanSave] = useState(false);
+    const [error, setError] = useState<ErrorText | undefined>();
+    const [updating, setUpdating] = useState(false);
+
+    const scrollViewRef = useRef<KeyboardAwareScrollView>();
+
+    const buttonText = intl.formatMessage({id: 'mobile.account.settings.save', defaultMessage: 'Save'});
+    const rightButton = useMemo(() => {
+        return isTablet ? null : {
+            id: 'update-profile',
+            enabled: false,
+            showAsAction: 'always' as const,
+            testID: 'edit_profile.save.button',
+            color: theme.sidebarHeaderTextColor,
+            text: buttonText,
+        };
+    }, [isTablet, theme.sidebarHeaderTextColor]);
+
+    useEffect(() => {
+        const unsubscribe = Navigation.events().registerComponentListener({
+            navigationButtonPressed: ({buttonId}: { buttonId: string }) => {
+                switch (buttonId) {
+                    case 'update-profile':
+                        submitUser();
+                        break;
+                    case closeButtonId:
+                        close();
+                        break;
+                }
+            },
+        }, componentId);
+
+        return () => {
+            unsubscribe.remove();
+        };
+    }, [userInfo]);
+
+    useEffect(() => {
+        const backHandler = BackHandler.addEventListener('hardwareBackPress', close);
+        return () => {
+            backHandler.remove();
+        };
+    }, []);
+
+    useEffect(() => {
+        if (!isTablet) {
+            setButtons(componentId, {rightButtons: [rightButton!]});
+        }
+    }, []);
+
+    const service = currentUser.authService;
+
+    const close = useCallback(() => {
+        if (isModal) {
+            dismissModal({componentId});
+        } else if (isTablet) {
+            DeviceEventEmitter.emit(Events.ACCOUNT_SELECT_TABLET_VIEW, '');
+        } else {
+            popTopScreen(componentId);
+        }
+
+        return true;
+    }, [isModal]);
+
+    const enableSaveButton = useCallback((value: boolean) => {
+        const buttons = {
+            rightButtons: [{...rightButton!, enabled: value}],
```

Comment:
It seems that you are calling enableSaveButton in places were you might not be in a tablet (I might be wrong), and therefore, rightButton may be null. Am I right? If so, maybe we shouldn't force the definition of right button with `!`.

Reply:
refactored